rlp/rlp.go

```diff
@@ -0,0 +1,43 @@
+package rlp
+
+import (
+	"encoding/binary"
+	"math"
+)
+
+type Item struct {
+	D []byte
+	L []*Item
+}
+
+func Encode(input *Item) []byte {
+	if input.D != nil {
```

Comment:
Ah so you avoid reflection by assuming that either `D` is present (ie it's a string) or if it's `nil`, then it's a list?

Reply:
Exactly! The insight I had was that by not accepting Go's standard data types we could make the internals much simpler. Instead, you must use `Item` to interact with this package. While this might lead to a little more coding, I think the simplicity is worth it.